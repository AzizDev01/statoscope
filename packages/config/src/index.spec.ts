import { requireConfig } from './index';

describe('requireConfig', () => {
  test('should work', () => {
    expect(requireConfig()).toMatchSnapshot();
    expect(requireConfig('fooooo')).toMatchSnapshot();
    expect(
      requireConfig(require.resolve('../../../test/fixtures/cli/validate/config'))
    ).toMatchSnapshot();
  });
});
