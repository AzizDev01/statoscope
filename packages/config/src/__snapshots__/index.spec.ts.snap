// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requireConfig should work 1`] = `
Object {
  "config": Object {},
  "path": "<pwd>/statoscope.config.js",
}
`;

exports[`requireConfig should work 2`] = `
Object {
  "config": Object {},
  "path": "<pwd>/fooooo",
}
`;

exports[`requireConfig should work 3`] = `
Object {
  "config": Object {
    "validate": Object {
      "plugins": Array [
        Array [
          "../../../../packages/stats-validator-plugin-webpack",
          "webpack",
        ],
      ],
      "rules": Object {
        "webpack/restricted-modules": Array [
          "error",
          Array [
            /\\\\/src\\\\//,
          ],
        ],
        "webpack/restricted-packages": Array [
          "error",
          Array [
            "foo",
          ],
        ],
      },
    },
  },
  "path": "<pwd>/test/fixtures/cli/validate/config.js",
}
`;
